flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/Buffer.java

```diff
@@ -227,4 +229,53 @@
 	 * Tags the buffer as compressed or uncompressed.
 	 */
 	void setCompressed(boolean isCompressed);
+
+	/**
+	 * Gets the type of data this buffer contains.
+	 */
+	DataType getDataType();
+
+	/**
+	 * Used to identify the type of data contained in {@link Buffer}.
```

Comment:
I notice that some usages are relying on the ordinal  of enum value. So do you think we should give some notes to avoid adjust the existing sequence if extending it future?

Reply:
The ordinal of enum value is used when writing buffer to disk and sending buffers through network. As suggested, I have removed those logic so the ordinal of enum value is never used now.